### PR TITLE
media-video/qmplay2: new ebuild

### PR DESCRIPTION
**Pull request by linxon**
A video and audio player. It can play all formats supported by `FFmpeg`, `libmodplug` (including J2B and SFX). It also supports Audio CD, raw files, Rayman 2 music and chiptunes. It contains YouTube and Prostopleer browser.

Homepage: https://github.com/zaps166/QMPlay2
ChangeLog: [19.12.19](https://github.com/zaps166/QMPlay2/releases/tag/19.12.19) 

Package-Manager: Portage-2.3.51, Repoman-2.3.11
Signed-off-by: Yury Martynov email@linxon.ru

### CODE REVIEW

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PLOCALES="de es fr hu pl ru uk zh"
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: Not needed

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+inherit cmake-utils l10n xdg
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: never use l10n anymore

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+MY_PN="QMPlay2"
+
+DESCRIPTION="Qt-based video player, which can play all formats and stream"
+HOMEPAGE="https://github.com/zaps166/${MY_PN}"
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: no ebuild variables in the HOMEPAGE, please

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	pulseaudio? ( media-sound/pulseaudio )
+	sid? ( media-libs/libsidplayfp )
+	taglib? ( media-libs/taglib )
+	vaapi? ( x11-libs/libva[opengl,X] )
+	vdpau? ( x11-libs/libvdpau )
+	xv? ( x11-libs/libXv )
+"
+DEPEND="${RDEPEND}
+	dev-qt/linguist-tools:5
+	virtual/pkgconfig
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: this is actually `BDEPEND` for EAPI=7

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+"
+
+CMAKE_MIN_VERSION="3.1"
+
+src_prepare() {
+	l10n_find_plocales_changes "${S}/lang" "" '.ts'
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: not needed

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+	sed -i -r \
+		-e 's/if\(GZIP\)/if\(TRUE\)/' \
+		-e 's/(install.+QMPlay2\.1)\.gz/\1/' \
+		src/gui/CMakeLists.txt
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: missing `|| die`

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	vaapi? ( x11-libs/libva[opengl,X] )
+	vdpau? ( x11-libs/libvdpau )
+	xv? ( x11-libs/libXv )
+"
+DEPEND="${RDEPEND}
+	dev-qt/linguist-tools:5
+	virtual/pkgconfig
+"
+
+CMAKE_MIN_VERSION="3.1"
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: why?

File: media-video/qmplay2/qmplay2-18.12.26.ebuild [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+	dbus? ( dev-qt/qtdbus:5 )
+	gme? ( media-libs/game-music-emu )
+	cdio? ( dev-libs/libcdio[cddb] )
+	jemalloc? ( dev-libs/jemalloc )
+	libass? ( media-libs/libass )
+	libav? ( media-video/libav:= )
+	!libav? ( >=media-video/ffmpeg-3.1:= )
+	mpris? ( dev-qt/qtdbus:5 )
+	portaudio? ( media-libs/portaudio )
+	pulseaudio? ( media-sound/pulseaudio )
```
Comment (Zlogene) [inline media-video/qmplay2/qmplay2-18.12.26.ebuild]: most of flags you provided here do not affect configure stage wither way, we do not accept RDEPEND only use flags, please remover all the flags that do not match `src_configure()`

Comment (linxon) [reply media-video/qmplay2/qmplay2-18.12.26.ebuild]: pulseaudio and other flags located above in to the MODULES=() array and checkable by "for" construction with in src_configure()

do i need to change something? Thank you!

**Comment by reagentoo** [comment]
Previous attempt of including to tree with @mgorny: https://github.com/gentoo/gentoo/pull/7772

**Comment by linxon** [comment]
@Zlogene bump

File: media-video/qmplay2/qmplay2-19.12.19.ebuild [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+LICENSE="LGPL-3"
+SLOT="0"
+
+IUSE="avdevice +audiofilters avresample alsa cdio cuvid dbus +extensions
+	gme inputs +lastfm libass lyrics mediabrowser modplug mpris2
+	notifications opengl portaudio pulseaudio sid svg taglib vaapi
+	vdpau +videofilters visualizations xv"
+
+REQUIRED_USE="
+	audiofilters? ( || ( alsa portaudio pulseaudio ) )
```
Comment (juippis) [inline media-video/qmplay2/qmplay2-19.12.19.ebuild]: You `+audiofilters` yet none of these, so the emerge stops right away asking user to enable some of these. You should enable something by default, like `+alsa`. 

... which I did and got
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindALSA.cmake:60 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/modules/ALSA/CMakeLists.txt:20 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/portage/media-video/qmplay2-19.12.19/work/qmplay2-19.12.19_build/CMakeFiles/CMakeOutput.log".
 * ERROR: media-video/qmplay2-19.12.19::testworld failed (configure phase):
 *   cmake failed
```

File: media-video/qmplay2/qmplay2-19.12.19.ebuild [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+	vaapi? ( opengl )"
+
+RDEPEND="
+	cdio? ( dev-libs/libcdio[cddb] )
+	dev-libs/jansson
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5[xcb]
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	svg? ( dev-qt/qtsvg:5 )
```
Comment (juippis) [inline media-video/qmplay2/qmplay2-19.12.19.ebuild]: `svg?` below sid please.

File: media-video/qmplay2/qmplay2-19.12.19.ebuild [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+	dbus? ( dev-qt/qtdbus:5 )
+	gme? ( media-libs/game-music-emu )
+	libass? ( media-libs/libass )
+	mediabrowser? ( dev-qt/qtdeclarative:5 )
+	>=media-video/ffmpeg-4.1.0[vaapi=,vdpau=]
```
Comment (juippis) [inline media-video/qmplay2/qmplay2-19.12.19.ebuild]: This is out of place as well.

**Comment by juippis** [comment]
Also please describe the package shortly in git message box with the commit.

**Comment by juippis** [comment]
Btw you can automate the sign-off part by adding `SIGNED_OFF_BY` in make.conf when you use repoman to commit. 

`man 5 make.conf`.

**Pull request closed without merging**